src/main/java/seedu/address/model/person/User.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents the user.
+ */
+public class User {
+
+    private Name name;
+    private Phone phone;
+    private Email email;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public User(Name name, Phone phone, Email email) {
+        requireAllNonNull(name, phone, email);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    public Email getEmail() {
+        return email;
+    }
+
+    /**
+     * Replaces the user with {@code newUser}.
+     */
+    public void setUser(User newUser) {
+        this.name = newUser.getName();
+        this.phone = newUser.getPhone();
+        this.email = newUser.getEmail();
+    }
+
+    /**
+     * Returns true if both users have the same data.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof User)) {
+            return false;
+        }
+
+        User otherUser = (User) other;
+        return otherUser.getName().equals(getName())
+                && otherUser.getPhone().equals(getPhone())
+                && otherUser.getEmail().equals(getEmail());
```

Comment:
Instead of returning the boolean directly, it might be better to refactor the code to explain it better:
```
boolean isSameData = otherUser.getName()..... && otherUser.getPh...... && ... ;
return isSameData;
```

Reply:
Erm.. for this, i just followed the implementation given for equal in other classes